lib/lwm2m_carrier/include/lwm2m_carrier.h

```diff
@@ -98,4 +193,181 @@ const char *lwm2m_carrier_timezone(int32_t tz_offset, int32_t dst);
+ *                                  of the LWM2M device.
+ *
+ * @return     -EINVAL              If the input argument is a NULL pointer or an empty string.
+ * @return     -E2BIG               If the input string is too long.
+ * @return     -ENOMEM              If it was not possible to allocate memory storage to hold the string.
+ * @return     0                    If the software version has been set successfully.
+ */
+int lwm2m_carrier_software_version_set(char * software_version);
+
+/**
+ * @brief      Update the device object instance error code by adding an individual error.
+ *
+ * @note       Upon initialisation of the device object instance, the error code is specified as 0,
+ *             indicating no error. The error code is to be updated whenever a new error occurs.
+ * @note       If the reported error is NO_ERROR, all existing error codes will be reset.
+ * @note       If the reported error is already present, the error code will remain unchanged.
+ *
+ * @param[in]  error                Individual error to be added.
+ *
+ * @return     -EINVAL              If the error code is not supported.
+ * @return     0                    If the error code has been added successfully.
+ */
+int lwm2m_carrier_error_code_add(lwm2m_carrier_error_code_t error);
+
+/**
+ * @brief      Update the device object instance error code by removing and individual error.
+ *
+ * @note       Upon initialisation of the device object instance, the error code is specified as 0,
+ *             indicating no error. The error code is to be updated whenever an error is no longer
+ *             present. When all the errors are removed, the error code is specified as 0, hence
+ *             indicating no error again.
+ *
+ * @param[in]  error            Individual error code to be removed.
+ *
+ * @return     -EINVAL          If the error code is not supported.
+ * @return     -ENOENT          If the error to be removed is not present on the error code list.
+ * @return     0                If the error has been removed successfully.
+ */
+int lwm2m_carrier_error_code_remove(lwm2m_carrier_error_code_t error);
+
+/**
+ * @brief      Set the total amount of storage space to store data and software in the LWM2M Device.
+ *
+ * @note       The value is expressed in kilobytes (kB).
+ *
+ * @param[in]  memory_total     Total amount of storage space in kilobytes.
+ *
+ * @return     -EINVAL          If the reported value of total amount of storage space is a negative
+ *                              value.
+ * @return     0                If the total amount of storage space has been set successfully.
+ */
+int lwm2m_carrier_memory_total_set(int32_t memory_total);
+
+/**
+ * @brief      Read the estimated current available amount of storage space to store data and
+ *             software in the LWM2M Device.
+ *
+ * @note       This function can be implemented by the application if needed.
```

Comment:
Who's the user of this function, why is the implementation optional (as I assume from `can be implemented`)?